connect/runtime/src/main/java/org/apache/kafka/connect/runtime/Worker.java

```diff
@@ -100,20 +102,39 @@ public Worker(
         this.config = config;
         this.workerMetricsGroup = new WorkerMetricsGroup(metrics);
 
-        // Internal converters are required properties, thus getClass won't return null.
-        this.internalKeyConverter = plugins.newConverter(
-                config.getClass(WorkerConfig.INTERNAL_KEY_CONVERTER_CLASS_CONFIG).getName(),
-                config
-        );
+        // Internal key/value converters are currently deprecated, instead JsonConverter is used by default, refer KIP-174.
+        // Currently Internal converters are required properties, thus getClass won't return null.
+
+        boolean isJsonConverterKeyClassUsed = config.getClass(WorkerConfig.INTERNAL_KEY_CONVERTER_CLASS_CONFIG).getName().equals("org.apache.kafka.connect.json.JsonConverter");
+        boolean isJsonConverterValueClassUsed = config.getClass(WorkerConfig.INTERNAL_VALUE_CONVERTER_CLASS_CONFIG).getName().equals("org.apache.kafka.connect.json.JsonConverter");
+
+        if (!isJsonConverterKeyClassUsed && !isJsonConverterValueClassUsed) {
+            this.internalKeyConverter =
+                    plugins.newConverter(config.getClass(WorkerConfig.INTERNAL_KEY_CONVERTER_CLASS_CONFIG).getName(), config);
+            this.internalValueConverter =
+                    plugins.newConverter(config.getClass(WorkerConfig.INTERNAL_VALUE_CONVERTER_CLASS_CONFIG).getName(), config);
+        } else {
+            this.internalKeyConverter = plugins.newConverter(JsonConverter.class.getName(), config);
+            this.internalValueConverter = plugins.newConverter(JsonConverter.class.getName(), config);
+        }
+
+        boolean isInternalConverterSchemaConfigured = config.originals().containsKey("internal.key.converter.schemas.enable") && config.originals().containsKey("internal.value.converter.schemas.enable");
```

Comment:
This is interesting and something I don't think we really considered with compatibility. Technically the default values only apply to the values in the ConfigDef, which means the actual default for `schemas.enable` is the default in `JsonConverter`, which is `true`, not the `false` value we setup in the example configs.

In practice I think making this change won't matter since I think offsets don't work with `JsonConverter` and `schemas.enable=true`, but it would be good to verify (and maybe even update the KIP compatibility discussion to clarify). There's a JIRA related to that issue which would be good justification for why we are pretty sure nobody is using it in that mode, I think it was closed due to this KIP but should be findable easily in JIRA search.

Reply:
Yes, I believe you are referring [KAFKA-3988](https://issues.apache.org/jira/browse/KAFKA-3988). Also we have already mentioned this in our KIP, so are you suggesting to edit the KIP to add something like: "KAFKA-3988 helped us to track the use of Converters with schema. Since we don't find any such usage so we assume that nobody is using Converters with schema." 

Let me know if I missed anything.